### PR TITLE
Publish `entities` frame

### PR DESCRIPTION
**Pull request by hakuturu583**
# Description

## Abstract

When I checking performance of scenario_simulator_v2, I found that it takes a lot of time for broadcasting frames for each entity.

## Background

When spawning a lot of entity (About 10), it takes a lot of times to broadcasting frames.
It causes frame drop, so we have to fix it.

## Details

Broadcast `entities` frame, which is based on the average position of all Entities.
This PR is depends on #1237 

## References

#1237 
https://github.com/tier4/sim_evaluation_tools/issues/274

# Destructive Changes

In previous version, all of the entity has tf frame and it was used in visualization, but it is removed in this pull request.

# Known Limitations

N/A

### CODE REVIEW

File: simulation/traffic_simulator/src/entity/entity_manager.cpp [hunk 1]
```diff
@@ -38,12 +39,27 @@ namespace entity
 void EntityManager::broadcastEntityTransform()
 {
   std::vector<std::string> names = getEntityNames();
-  for (const auto & name : names) {
-    geometry_msgs::msg::PoseStamped pose;
-    pose.pose = getMapPose(name);
-    pose.header.stamp = clock_ptr_->now();
-    pose.header.frame_id = name;
-    broadcastTransform(pose);
+  if (entityExists("ego")) {
```
Comment (HansRobo) [inline simulation/traffic_simulator/src/entity/entity_manager.cpp:42]: Please do not use "ego" directly and use `getEgoName` instead.

Comment (HansRobo) [reply simulation/traffic_simulator/src/entity/entity_manager.cpp:42]: But this is just a nit, and does not block merging.

File: simulation/traffic_simulator/src/entity/entity_manager.cpp [hunk 2]
```diff
@@ -38,12 +39,27 @@ namespace entity
...
+      geometry_msgs::build<geometry_msgs::msg::PoseStamped>()
+        .header(
+          std_msgs::build<std_msgs::msg::Header>().stamp(clock_ptr_->now()).frame_id("entities"))
+        .pose(geometry_msgs::build<geometry_msgs::msg::Pose>()
+                .position(std::accumulate(
+                  names.begin(), names.end(), geometry_msgs::msg::Point(),
+                  [this, names](geometry_msgs::msg::Point & point, const std::string & name) {
+                    return point +
+                           (getMapPose(name).position * (1.0 / static_cast<double>(names.size())));
+                  }))
```
Comment (HansRobo) [inline simulation/traffic_simulator/src/entity/entity_manager.cpp:60]: What is this calculation for?

Comment (hakuturu583) [reply simulation/traffic_simulator/src/entity/entity_manager.cpp:60]: This part of the process is intended to ensure that frames are issued in a position that makes it as easy as possible to see the entities that will appear in the scenario.

Comment (hakuturu583) [reply simulation/traffic_simulator/src/entity/entity_manager.cpp:60]: In the past, we used to publish the frames of all entities, but that would be too heavy processing, so we publish the average of the coordinates of all entities.

File: simulation/traffic_simulator/src/entity/entity_manager.cpp [hunk 3]
```diff
@@ -38,12 +39,30 @@ namespace entity
...
-    pose.pose = getMapPose(name);
-    pose.header.stamp = clock_ptr_->now();
-    pose.header.frame_id = name;
-    broadcastTransform(pose);
+  if (isEgoSpawned()) {
+    const auto ego_name = getEgoName();
+    if (entityExists(ego_name)) {
+      broadcastTransform(
+        geometry_msgs::build<geometry_msgs::msg::PoseStamped>()
+          .header(std_msgs::build<std_msgs::msg::Header>().stamp(clock_ptr_->now()).frame_id("ego"))
```
Comment (HansRobo) [inline simulation/traffic_simulator/src/entity/entity_manager.cpp:47]:
```suggestion
          .header(std_msgs::build<std_msgs::msg::Header>().stamp(clock_ptr_->now()).frame_id(ego_name))
```

File: openscenario/openscenario_interpreter/src/openscenario_interpreter.cpp [hunk 4]
```diff
@@ -162,7 +162,7 @@ auto Interpreter::on_activate(const rclcpp_lifecycle::State &) -> Result
   auto evaluate_storyboard = [this]() {
     withExceptionHandler(
       [this](auto &&...) {
-        publishCurrentContext();
+        // publishCurrentContext();
```
Comment (HansRobo) [inline openscenario/openscenario_interpreter/src/openscenario_interpreter.cpp:165]: Why this line is commented out?

Comment (hakuturu583) [reply openscenario/openscenario_interpreter/src/openscenario_interpreter.cpp]: Sorry, I fixed this problem.

File: simulation/traffic_simulator/src/entity/entity_manager.cpp (hunk 3 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/entity/entity_manager.cpp:53]: This is the intended implementation.
It is easier to create rviz config if the name “ego” is fixed, so the frame_id “ego” is issued regardless of the name of the ego entity.

Comment (HansRobo) [reply simulation/traffic_simulator/src/entity/entity_manager.cpp:53]: I see, I understand.

File: simulation/traffic_simulator/src/entity/entity_manager.cpp (hunk 2 above)
Comment (HansRobo) [reply simulation/traffic_simulator/src/entity/entity_manager.cpp:74]: Clearly documented by https://github.com/tier4/scenario_simulator_v2/pull/1238/commits/0fe90566ed68e5e13710eec37bbf6ac769cda0ee

File: simulation/traffic_simulator/src/entity/entity_manager.cpp (hunk 1 above)
Comment (HansRobo) [reply simulation/traffic_simulator/src/entity/entity_manager.cpp]: Fixed by https://github.com/tier4/scenario_simulator_v2/pull/1238/commits/61af6463271fb24640a32fbb075e921d31677cb0

File: simulation/traffic_simulator/src/entity/entity_manager.cpp (hunk 3 above)
Comment (HansRobo) [reply simulation/traffic_simulator/src/entity/entity_manager.cpp:58]: I checked well-documented in https://github.com/tier4/scenario_simulator_v2/pull/1238/commits/9b5444900e2ac529131c5bf1ab5d4481465baf49

**Review by HansRobo — APPROVED** [review]

**Pull request merged**